engine/access/rpc/backend/connection_factory.go

```diff
@@ -28,7 +35,8 @@ func (cf *ConnectionFactoryImpl) createConnection(address string) (*grpc.ClientC
 	conn, err := grpc.Dial(
 		address,
 		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
-		grpc.WithInsecure())
+		grpc.WithInsecure(),
+		withClientUnaryInterceptor())
```

Comment:
should this be taking in the timeout as an option, rather than taking it directly from the global? or does that make something else very difficult

Reply:
nope..it can definitely be a parameter. I can change it be that.